### ISSUE TITLE
OpenCL calls broken in 10.0.2

### ISSUE THREAD

**Issue by guihomework** [iclcv/icl#37]
In 10.0.2 using OpenCL created these issues
```
ERROR: clEnqueueWriteBuffer clErrorCode -36
ERROR: clEnqueueNDRangeKernel clErrorCode -36
ERROR: clEnqueueReadBuffer clErrorCode -36
```

these were tracked down to the "throw" clean-up commit 86480c5f015eff0600d6c572a8661d2910a533d6 that unfortunately has "eaten up" some mandatory cmdQueue initialization in CLBuffer and CLKernel 
here https://github.com/iclcv/icl/commit/86480c5f015eff0600d6c572a8661d2910a533d6#diff-e0563068c6edf61678b0902384130972L76
and here
https://github.com/iclcv/icl/commit/86480c5f015eff0600d6c572a8661d2910a533d6#diff-618ef9bd1e6c0ebc09f73e62ac7eb9b2L49

a PR will be done with a fix

there might be other "eaten-up" commands depending if the clean-up was automatized and ignored any commands after the closing parenthesis of the throw.

**Comment by rhaschke** [comment]
@aleneum, do you remember which (sed?) command you used to cleanup the source?
Maybe, we should have another check to validate that you didn't erase to much...

**Comment by aleneum** [comment]
My regex was probably too greedy. Looking at the error I guess I forgot to prevent closing (and reopening) brackets INSIDE the brackets.

I copied the whole diff into a file
`git diff 86480c5f015eff0600d6c572a8661d2910a533d6~ 86480c5f015eff0600d6c572a8661d2910a533d6 > diff.txt`

and checked for deleted lines (starting with `-`) that included single colons: `^-.*[^:]:[^:]`.

I found 15 issues:

```
grep ^-.*[^:]:[^:] diff.txt
-    PseudoColorConverter::PseudoColorConverter(const std::vector<Stop> &stops, int maxValue) throw (ICLException):m_data(new Data){
-        GenericGrabber(const utils::ProgArg &pa) throw (utils::ICLException):m_poGrabber(0),m_remoteServer(0){
-                       bool notifyErrors = true) throw (utils::ICLException):m_poGrabber(0),m_remoteServer(0){
-    OpenCVCamGrabber::OpenCVCamGrabber(int dev)  throw (ICLException) :device(dev),m_buffer(0){
-    throw (FileNotFoundException) : data(new Data), mutex(Mutex::mutexTypeRecursive), updating(false){
-                                 Mode mode) throw(utils::ICLException) : m_data(new Data){
-    SwissRangerGrabber::SwissRangerGrabber(int serialNumber, depth bufferDepth, int pickChannel) throw (ICLException):Grabber(){
-    XiGrabber::XiGrabber(int deviceID) throw(utils::ICLException) : m_data(0){
-    ZmqGrabber::ZmqGrabber(const std::string &host, int port) throw(utils::ICLException):m_data(0){
-      throw (utils::ICLException):MarkerGridDetector::GridDefinition(numCells,markerIDs, markerType){
-                                                       bool deepCopyCams) throw (ICLException):m_data(0){
-    TwoLevelRegionStructure::TwoLevelRegionStructure(const std::string &code) throw (ICLException) : color(0), code(code){
-      Prop(const utils::Configurable *cfg) throw (utils::ICLException) : GUIComponent("prop","@pointer@:"+utils::Any::ptr(cfg)){
-        throw (CLBufferException):cmdQueue(cmdQueue) {
-        throw (CLKernelException):cmdQueue(cmdQueue) {
```

I will fix the remaining issues in `fix37`.
If you want to do some checks yourself you can have a look at the attached [diff.txt](https://github.com/iclcv/icl/files/3353692/diff.txt).

**Comment by guihomework** [comment]
I went through the diff manually, rather quickly, to look if there is no other type of things that would have been eaten up (other than after a ':' ) and did not find any.

**Comment by guihomework** [comment]
Do not hesitate to "throw" away my PR and do a clean one, since this would be more consistent than a manual fix, and the error logging can be handled by #40 later.

**Comment by aleneum** [comment]
closed in favor of #39

**Comment by rhaschke** [comment]
To maintain @guihomework improvements of the logging, I filed #42 using `ERROR_LOG` macros.
